imagetest/test_suites/image_boot/image_boot_test.go

```diff
@@ -189,65 +282,36 @@ func TestStartTime(t *testing.T) {
 }
 
 func TestBootTime(t *testing.T) {
-	image, err := utils.GetMetadata("image")
-	if err != nil {
-		t.Fatalf("couldn't get image from metadata")
-	}
-
-	var maxThreshold int
-
-	switch {
-	case strings.Contains(image, "almalinux"):
-		maxThreshold = imageFamilyBootTimeThresholdMap["almalinux"]
-	case strings.Contains(image, "centos"):
-		maxThreshold = imageFamilyBootTimeThresholdMap["centos"]
-	case strings.Contains(image, "debian"):
-		maxThreshold = imageFamilyBootTimeThresholdMap["debian"]
-	case strings.Contains(image, "rhel"):
-		maxThreshold = imageFamilyBootTimeThresholdMap["rhel"]
-	case strings.Contains(image, "rocky-linux"):
-		maxThreshold = imageFamilyBootTimeThresholdMap["rocky-linux"]
-	case strings.Contains(image, "sles-12"):
-		maxThreshold = imageFamilyBootTimeThresholdMap["sles-12"]
-	case strings.Contains(image, "sles-15"):
-		maxThreshold = imageFamilyBootTimeThresholdMap["sles-15"]
-	case strings.Contains(image, "ubuntu-pro"):
-		maxThreshold = imageFamilyBootTimeThresholdMap["ubuntu-pro"]
-	case strings.Contains(image, "ubuntu"):
-		maxThreshold = imageFamilyBootTimeThresholdMap["ubuntu"]
-	default:
-		t.Fatalf("OS not supported for boot time test")
-	}
 
-	var foundGuestAgentAndSshd bool
+	var foundPassCondition bool
 
-	// 120 is the current maximum number of seconds to allow any distro to start sshd and guest-agent before returning a test failure
-	for i := 0; i < 120; i++ {
-		if err := lookForSshdAndGuestAgentProcess(); err == nil {
-			foundGuestAgentAndSshd = true
+	// 300 is the current maximum number of seconds to allow any distro to start sshd and guest-agent before returning a test failure
+	for i := 0; i < 300; i++ {
+		if runtime.GOOS == "windows" {
+			if err := lookForGuestAgentProcessWindows(); err == nil {
+				foundPassCondition = true
+				break
+			}
+		} else if err := lookForSshdAndGuestAgentProcess(); err == nil {
+			foundPassCondition = true
 			break
 		}
 		time.Sleep(1 * time.Second)
 	}
 
-	if !foundGuestAgentAndSshd {
-		t.Fatalf("Condition for guest agent and sshd process to start not reached within timeout")
+	if !foundPassCondition {
+		t.Fatalf("Condition for guest agent and/or sshd process to start not reached within timeout")
 	}
 
-	// Reading the system uptime once both guest agent and sshd are found in the processes
-	uptimeData, err := os.ReadFile("/proc/uptime")
-	if err != nil {
-		t.Fatalf("Failed to read uptime file")
-	}
-	fields := strings.Split(string(uptimeData), " ")
-	uptime, err := strconv.ParseFloat(fields[0], 64)
-	if err != nil {
-		t.Fatalf("Failed to read uptime numeric value")
-	}
-	t.Logf("found guest agent and sshd running at %v seconds", int(uptime))
-
-	//Validating the uptime against the allowed threshold value
-	if int(uptime) > maxThreshold {
-		t.Errorf("Boot time too long: %v is beyond max of %v", uptime, maxThreshold)
+	if runtime.GOOS == "windows" {
```

Comment:
This conditional comes up a lot through out these tests. Can we have a helper test function that returns a boolean if the runtime is windows so that there is less of a chance to have a typo?

Reply:
I'm thinking I'll add an IsWindows() function to our utils because this is going to come up alot more in other test suites, yeah.